src/base/bittorrent/session.cpp

```diff
@@ -4404,33 +4531,33 @@ void Session::handleSessionStatsAlert(libt::session_stats_alert *p)
     m_status.trackerDownloadRate = calcRate(m_status.trackerDownload, trackerDownload);
     m_status.trackerUploadRate = calcRate(m_status.trackerUpload, trackerUpload);
 
-    m_status.totalDownload = totalDownload;
-    m_status.totalUpload = totalUpload;
-    m_status.totalPayloadDownload = totalPayloadDownload;
-    m_status.totalPayloadUpload = totalPayloadUpload;
-    m_status.ipOverheadDownload = ipOverheadDownload;
-    m_status.ipOverheadUpload = ipOverheadUpload;
-    m_status.trackerDownload = trackerDownload;
-    m_status.trackerUpload = trackerUpload;
-    m_status.dhtDownload = dhtDownload;
-    m_status.dhtUpload = dhtUpload;
-    m_status.totalWasted = p->values[m_metricIndices.net.recvRedundantBytes]
-            + p->values[m_metricIndices.net.recvFailedBytes];
-    m_status.dhtNodes = p->values[m_metricIndices.dht.dhtNodes];
-    m_status.diskReadQueue = p->values[m_metricIndices.peer.numPeersUpDisk];
-    m_status.diskWriteQueue = p->values[m_metricIndices.peer.numPeersDownDisk];
-    m_status.peersCount = p->values[m_metricIndices.peer.numPeersConnected];
-
-    const int numBlocksRead = p->values[m_metricIndices.disk.numBlocksRead];
-    const int numBlocksCacheHits = p->values[m_metricIndices.disk.numBlocksCacheHits];
-    m_cacheStatus.totalUsedBuffers = p->values[m_metricIndices.disk.diskBlocksInUse];
+    m_status.totalDownload = boost::numeric_cast<quint64>(totalDownload);
```

Comment:
Is counters type changed in libtorrent-1.2? Why is `boost::numeric_cast<quint64>` needed? Are you afraid it might not fit in `quint64`?

Reply:
I imagine there may be a conversion warning here, because the libtorrent counters are signed.

Reply:
(which I think your counters should be too)

Reply:
If all supported libtorrent versions have the counters of signed type we should use qint64.

Reply:
unfortunately 1.1.x uses `std::uint64_t`